### PR TITLE
Upgrade to C# 13

### PR DESCRIPTION
**Pull request by westin-m**
Increment to C# 13
Fix undeclared API warnings

### CODE REVIEW

**Review by pmaytak — APPROVED** [review]

**Comment by jennyf19** [comment]
we might want to update all the Id web one's to 13: https://github.com/search?q=repo%3AAzureAD%2Fmicrosoft-identity-web%20LangVersion&type=code

File: src/Microsoft.Identity.Web.TokenAcquisition/PublicAPI/net9.0/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #nullable enable
+const Microsoft.Identity.Web.ClaimConstants.Password = "xms_password" -> string!
```
Comment (msbw2) [inline src/Microsoft.Identity.Web.TokenAcquisition/PublicAPI/net9.0/PublicAPI.Unshipped.txt:2]: What are these changes?

Comment (westin-m) [reply src/Microsoft.Identity.Web.TokenAcquisition/PublicAPI/net9.0/PublicAPI.Unshipped.txt:2]: #3140 introduced these but it wasn't built with net 9, so the associated warnings didn't show up

**Review by pmaytak — COMMENTED** [review]
Looks like more places need to be updated: https://github.com/search?q=repo%3AAzureAD%2Fmicrosoft-identity-web+%22%3CLangVersion%3E%22&type=code

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by jennyf19** [comment]
please create an issue as well. thanks.

**Comment by westin-m** [comment]
> please create an issue as well. thanks.

The one you created is already linked :)

**Pull request merged**